File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/DeleteUserUseCase.java
package br.org.restaurantepopular.usecase.user;

import br.org.restaurantepopular.usecase.core.IUseCase;

import java.util.Optional;

public class DeleteUserUseCase implements IUseCase<String, Optional> {

    private final IUserRepository userRepository;

    public DeleteUserUseCase(IUserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public Optional execute(String s) {
        userRepository.delete(s);
        return Optional.empty();
    }
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/DeleteUserUseCase.kt
package br.org.restaurantepopular.usecase.user

import br.org.restaurantepopular.usecase.core.IUseCase

class DeleteUserUseCase(
    private val userRepository: IUserRepository
) : IUseCase<String, Unit> {

    override fun execute(request: String) {
        userRepository.delete(request)
    }
}